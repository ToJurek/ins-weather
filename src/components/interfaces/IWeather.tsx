export default interface IWeather {
    icon: string;
    description: string;
    temp: string;
    feelsLike: string;
    tempMin: string;
    tempMax: string;
    pressure: string;
    humidity: string;
    clouds: string;
    time: string
}
