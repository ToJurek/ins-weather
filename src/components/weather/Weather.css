.App .shadow-background {
    width: 25%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
}

.shadow-background .card {
    font-family: 'Calibri';
    font-size: 25px;
    width: 90%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    background: hsla(0, 0%, 5%, 0.41);
    color: #f0f5f5;
}

content {
    width: 90%;
    height: 50%;
    border-radius: 2px;
}

.dashboard {
    display: block;
}

@media only screen and (max-width: 950px) {
    .App .shadow-background {
        width: 70%;
    }
}

@media only screen and (max-width: 600px) {
    .App .shadow-background {
        width: 100%;
    }
}
