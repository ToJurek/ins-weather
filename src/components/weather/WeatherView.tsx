import React from 'react';
import './Weather.css';
import IWeather from "../interfaces/IWeather";
import Header from "../cardHeader/Header";
import Body from "../cardBody/Body";
import Button from "../button/Button";

interface Props {
    weather: IWeather,
    refetchData: any
}

function WeatherView(props: Props) {
    let weather: IWeather = props.weather;
    return (
        <div className={'content'}>
            <div className={"dashboard"}>
                <Header city={"WrocÅ‚aw"}/>
                <Body weather={weather}/>
            </div>
            <Button handleClick={props.refetchData} name={"Update"}/>
        </div>
    )

}

export default WeatherView;
