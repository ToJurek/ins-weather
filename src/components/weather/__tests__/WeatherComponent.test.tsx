import React from "react";
import Enzyme, {mount, shallow} from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import Weather from "../WeatherComponent";
import WeatherView from "../WeatherView";
import IWeather from "../../interfaces/IWeather";
Enzyme.configure({adapter: new Adapter()});

describe('WeatherComponent', function () {
    test('should match to snapshot', () => {
        let wrapper = mount(<Weather/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    test('should render child', () => {
        let wrapper = shallow(<Weather/>)
        expect(wrapper.find(WeatherView)).toBeTruthy()
    })
});
