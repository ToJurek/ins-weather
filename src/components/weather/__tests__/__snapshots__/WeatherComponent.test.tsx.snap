// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherComponent should match to snapshot 1`] = `
<Weather>
  <div
    className="shadow-background"
  >
    <div
      className="card"
    >
      <State
        error=""
        isLoaded={false}
      >
        <div
          className="spinner"
        >
          <div
            className="bounce1"
          />
          <div
            className="bounce2"
          />
          <div
            className="bounce3"
          />
        </div>
      </State>
    </div>
  </div>
</Weather>
`;
